doc/howtos/profilecode.rst

```diff
@@ -0,0 +1,89 @@
+===================
+Profiling Odoo code
+===================
+
+.. warning::
+
+    This tutorial requires :ref:`having installed Odoo <setup/install>`
+    and :ref:`writing Odoo code <backend>`
+
+Graph a method
+==============
+
+Odoo embeds a profiler of code. This embeded profiler output can be used to
+generate a graph of calls triggered by the method, percentage of time taken in
+the method itself as well as time taken in method and it's sub-called methods.
+
+.. code:: python
+
+    from odoo.tools.misc import profile
+    [...]
+    @profile('/temp/prof.profile')
+    @api.multi
+    def mymethod(...)
+
+This produce a file called /temp/prof.profile
+
+A tool called *gprof2dot* will produce a graph with this result:
+
+.. code:: bash
+
+    gprof2dot -f pstats -o /temp/prof.xdot /temp/prof.profile
+
+A tool called *xdot* will display the resulting graph:
+
+.. code:: bash
+    
+    xdot /temp/prof.xdot
+    
+Dump stack
+==========
+
+Sending the SIGQUIT signal to an odoo process (only available on POSIX) makes
+this process output the current stack trace to log, with info level. When an
+odoo process seems stucked, sending this signal to the process permit to know
+what the process is doing, and letting the process continue his job.
+
+Tracing code execution
+======================
+
+Instead of sending the SIGQUIT signal to an odoo process often enough, to check
+where processes is performing worse than expected, we can use pyflame tool to
+do it for us.
+
+Install pyflame and flamegraph
+------------------------------
+
+.. code:: bash
+
+    # These instructions are given for Debian/Ubuntu distributions
+    sudo apt install autoconf automake autotools-dev g++ pkg-config python-dev python3-dev libtool make
+    git clone https://github.com/uber/pyflame.git
+    git clone https://github.com/brendangregg/FlameGraph.git
+    cd pyflame
+    ./autogen.sh
+    ./configure
+    make
+    sudo make install
```

Comment:
There are simpler alternatives, but that's not the point.
The point of this documentation is *how to use pyflame with odoo*.
For the installation of `pyflame`, simply refer to its documentation, https://pyflame.readthedocs.io